src/cryptographic_primitives/proofs/sigma_correct_homomorphic_elgamal_enc.rs

```diff
@@ -21,56 +19,59 @@ use zeroize::Zeroize;
 /// The relation R outputs 1 if D = xH+rY , E = rG (for the case of G=H this is ElGamal)
 ///
 #[derive(Clone, PartialEq, Debug, Serialize, Deserialize)]
-pub struct HomoELGamalProof {
-    pub T: GE,
-    pub A3: GE,
-    pub z1: FE,
-    pub z2: FE,
+pub struct HomoELGamalProof<P: ECPoint> {
+    pub T: P,
+    pub A3: P,
+    pub z1: P::Scalar,
+    pub z2: P::Scalar,
 }
 
 #[derive(Clone, PartialEq, Debug, Serialize, Deserialize)]
-pub struct HomoElGamalWitness {
-    pub r: FE,
-    pub x: FE,
+pub struct HomoElGamalWitness<S> {
```

Comment:
shouldn't `S` here be forced to be P:Scalar ?

Reply:
You're right. Just fixed it